cdk/ecs/lib/stacks/ecsStack.ts

```diff
@@ -0,0 +1,1118 @@
+            image: ContainerImage.fromRegistry(
+                `${this.ecrImagePrefix}/springcommunity/spring-petclinic-discovery-server`,
+            ),
+            cpu: 256,
+            memoryLimitMiB: 512,
+            essential: true,
+            environment: {
+                SPRING_PROFILES_ACTIVE: 'ecs',
+                CONFIG_SERVER_URL: `http://${this.CONFIG_SERVER}-DNS.${this.serviceDiscoveryStack.namespace.namespaceName}:8888`,
+            },
+            logging: LogDrivers.awsLogs({
+                streamPrefix: 'ecs',
+                logGroup: discoveryLogGroup,
+            }),
+        });
+
+        // Add Port Mapping
+        container.addPortMappings({
+            containerPort: 8761,
+            protocol: Protocol.TCP,
+        });
+
+        // Create ECS service
+        this.createService({
+            serviceName: this.DISCOVERY_SERVER,
+            taskDefinition: taskDefinition,
+        });
+    }
+
+    createAdminServer() {
+        // Create a log group
+        const adminLogGroup = this.logStack.createLogGroup(this.ADMIN_SERVER);
+
+        // Create ECS task definition
+        const taskDefinition = new TaskDefinition(this, `${this.ADMIN_SERVER}-task`, {
+            cpu: '256',
+            memoryMiB: '512',
+            compatibility: Compatibility.FARGATE,
+            family: this.ADMIN_SERVER,
+            networkMode: NetworkMode.AWS_VPC,
+            taskRole: this.ecsTaskRole,
+            executionRole: this.ecsTaskExecutionRole,
+        });
+
+        // Add Container to Task Definition
+        const container = taskDefinition.addContainer(`${this.ADMIN_SERVER}-container`, {
+            image: ContainerImage.fromRegistry(`${this.ecrImagePrefix}/springcommunity/spring-petclinic-admin-server`),
+            cpu: 256,
+            memoryLimitMiB: 512,
+            essential: true,
+            environment: {
+                SPRING_PROFILES_ACTIVE: 'ecs',
+                CONFIG_SERVER_URL: `http://${this.CONFIG_SERVER}-DNS.${this.serviceDiscoveryStack.namespace.namespaceName}:8888`,
+                DISCOVERY_SERVER_URL: `http://${this.DISCOVERY_SERVER}-DNS.${this.serviceDiscoveryStack.namespace.namespaceName}:8761/eureka`,
+                ADMIN_IP: `${this.ADMIN_SERVER}-DNS.${this.serviceDiscoveryStack.namespace.namespaceName}`,
+            },
+            logging: LogDrivers.awsLogs({
+                streamPrefix: 'ecs',
+                logGroup: adminLogGroup,
+            }),
+        });
+
+        // Add Port Mapping
+        container.addPortMappings({
+            containerPort: 9090,
+            protocol: Protocol.TCP,
+        });
+
+        // Create ECS service
+        this.createService({
+            serviceName: this.ADMIN_SERVER,
+            taskDefinition: taskDefinition,
+        });
+    }
+
+    createApiGateway(loadBalancerDNS: string, adotJavaImageTag: string, targetGroup: ApplicationTargetGroup) {
```

Comment:
There seems to be a lot of duplicated code in function `createApiGateway`, `runVetsService`, `runCustomersService`, and `runVisitsService`. Can we refactor to extract the common part out?